cfn-resources/test/e2e/search-deployment/searchdeployment_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package searchdeployment_test
+
+import (
+	ctx "context"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/test/e2e/utility"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+type localTestContext struct {
+	cfnClient      *cloudformation.Client
+	atlasClient    *admin.APIClient
+	projectTmplObj testProject
+	resourceCtx    utility.ResourceContext
+	template       string
+}
+
+type testProject struct {
+	ResourceTypeName string
+	Profile          string
+	ProjectID        string
+	ClusterName      string
+}
```

Comment:
Could we add InstanceSize and NodeCount here? Leave as dynamic value in the stack template, and then we can assert the values in the `testCreateStack`

Reply:
InstanceSize for sure, i had a problem with NodeCount because it's an int and the JSON was bad formed, let's discuss offline

Reply:
updated